docs/admin/prometheus.md

```diff
@@ -35,8 +35,9 @@ The Prometheus endpoint can be enabled in the
 [Helm chart's](https://github.com/coder/coder/tree/main/helm) `values.yml` by
 setting the environment variable `CODER_PROMETHEUS_ADDRESS` to `0.0.0.0:2112`.
 The environment variable `CODER_PROMETHEUS_ENABLE` will be enabled
-automatically. A Service Endpoint will also be exposed allowing Prometheus
-Service Monitors to be used.
+automatically. A Service Endpoint will not be exposed; if you need to expose the
+Prometheus port on a Service, (for example, to use a `ServiceMonitor`), create a
+separate service instead.
```

Comment:
IIUC, you don't need to have the prometheus port on the service to use a ServiceMonitor, [because you can directly set the targetPort](https://observability.thomasriley.co.uk/prometheus/configuring-prometheus/using-service-monitors/).

Reply:
I'm reading reports that in this case, you may need to define the [Endpoints](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/endpoints-v1/) explicitly.

Reply:
Added example headless service spec